src/plugins/intel_npu/src/plugin/npuw/attention.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "attention.hpp"
+
+#include "util.hpp"
+
+ov::npuw::runtime::attention::PositionIDs::PositionIDs(std::size_t param_idx,
+                                                       const ov::npuw::compiled::Attention& d,
+                                                       const ov::ISyncInferRequest& rq)
+    : m_position_ids_idx(param_idx),
+      m_d(d),
+      m_rq(rq) {
+    // FIXME: speculative decode is indistinguishable at this point!
+    m_case = m_d.query_size == 1 ? Case::GENERATE : Case::PREFILL;
+}
+
+ov::npuw::runtime::attention::Selector::Ptr ov::npuw::runtime::attention::PositionIDs::find(
+    const ov::npuw::compiled::Attention& d,
+    const ov::ISyncInferRequest& rq) {
+    auto is_position_ids = [](const ov::Output<const ov::Node>& p) {
+        const auto& shape = p.get_shape();
+        // FIXME: 2D/3D position IDs are not supported here YET
+        return p.get_node()->get_friendly_name() == "position_ids" &&
+               (shape.size() == 1 || (shape.size() == 2 && shape[0] == 1));
+    };
+
+    const auto& inputs = rq.get_inputs();
+    auto pos_ids_iter = std::find_if(inputs.begin(), inputs.end(), is_position_ids);
+    if (pos_ids_iter != inputs.end()) {
+        const auto param_idx = std::distance(inputs.begin(), pos_ids_iter);
+        return Selector::Ptr{new PositionIDs(param_idx, d, rq)};
```

Comment:
I'd either rename `Ptr` to `SPtr` or `SoPtr`, or use `make_shared`. Otherwise it looks like we manage this memory

Reply:
`Ptr` was just a bit confusing to me. But you can ignore this comment